R/vegawidget.R

```diff
@@ -259,27 +259,34 @@ vegawidgetOutput <- function(outputId, width = "auto", height = "auto",
 #'
 #' @param expr expression that generates a vegawidget. This can be
 #'   a `vegawidget` or a `vegaspec`.
-#' @param env The environment in which to evaluate \code{expr}.
-#' @param quoted Is \code{expr} a quoted expression (with \code{quote()})? This
-#'   is useful if you want to save an expression in a variable.
+#' @param env **Deprecated**. The environment in which to evaluate \code{expr}.
+#' @param quoted **Deprecated**.
+#'   Is \code{expr} a quoted expression (with \code{quote()})?
+#'   This is useful if you want to save an expression in a variable.
 #'
 #' @export
 #'
-renderVegawidget <- function(expr, env = parent.frame(), quoted = FALSE) {
+renderVegawidget <- function(expr, env, quoted) {
 
   assert_packages("shiny")
 
+  if (lifecycle::is_present(env)) {
+    lifecycle::deprecate_warn("0.4", "vegawidget::renderVegawidget(env)")
+  }
+
+  if (lifecycle::is_present(quoted)) {
+    lifecycle::deprecate_warn("0.4", "vegawidget::renderVegawidget(quoted)")
+  }
+
   # if sent a vegaspec, convert to a vegawidget
   if (inherits(expr, "vegaspec")) {
     expr <- vegawidget(expr)
   }
```

Comment:
I suspect this is a real issue that needs to be addressed. This will force the user's expression to evaluate before getting passed to `shinyRenderWidget()` (i.e., `shinyRenderWidget()` is getting passed a `vegawidget()` object, not an expression)

I think what you really want here is to _wrap_ the user's expression in `function(x) if inherits(x, "vegaspec") vegawidget(x) else x` without evaluating it. `plotly::renderPlotly()` actually does the same thing (wraps the user expression with `plotly:::prepareWidget()`), so you could basically copy that same approach (BTW, I wouldn't worry about deprecating `quote`/`env` for now)

Reply:
Thanks Carson, I'll give it a shot!

Reply:
This works! Hoping I got the gist of the conversation.